myia_backend_pytorch/scripts/update_env.py

```diff
@@ -0,0 +1,55 @@
+import argparse
+
+from yaml import dump, load
+
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except ImportError:
+    from yaml import Loader, Dumper
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("file", help="YAML file path")
+    parser.add_argument(
+        "-p",
+        "--packages",
+        help="File containing packages requirements " "to add to YAML file.",
+    )
+    parser.add_argument(
+        "-o", "--output", help="Output YAML file. By default, print output."
+    )
+    args = parser.parse_args()
+    with open(args.file) as file:
+        data = load(file, Loader=Loader)
+    if args.packages:
+        packages = []
+        with open(args.packages) as file:
+            for line in file:
+                line = line.strip()
+                if not line:
+                    continue
+                packages.append(line)
+        dependencies = data.get("dependencies", [])
+        packages_added = False
+        for index, element in enumerate(dependencies):
+            if isinstance(element, dict):
+                dependencies = (
+                    dependencies[:index] + packages + dependencies[index:]
+                )
+                packages_added = True
+                break
+        if not packages_added:
+            dependencies.extend(packages)
+        data["dependencies"] = dependencies
+    data.pop("name", None)
```

Comment:
Why do you remove the env name for the update?

Reply:
When I keep the name, the script produces following env output for relay backend:
```
dependencies:
- python>=3.7,<4.0
- numpy>=1.16,<2.0
- scipy>=1.5,<2.0
- decorator>=4,<5
- attrs>=19,<20
- pip
- mila-iqia::tvm=0.7dev1+5.*
- pip:
  - antlr4-python3-runtime==4.7.2
name: myia_backend_relay
```

The name is at the end, and, I don't know why, but tvm package is never installed in such case.
When I remove the name field, everything is installed correctly.

I guess name field is not mandatory, so I think I can remove it without issues.

Reply:
It's at the end because the yaml dumper sorts keys by default.  You can disable that with `sort_keys=False`.  If you have another reason to remove it though, it's fine.

Reply:
OK, I make further checks, and it appears that, if name is included, conda will create an environment with specified name instead of installing packages in current activated environment. Thus, if specified name is not current environment name, then current environment won't get updated.

So, I must keep removing the name field!

```
(myia) notoraptor@notoraptor-linux:~/mila/dev/git/myia/myia_backend_relay$ conda info --envs
# conda environments:
#
base                     /home/notoraptor/anaconda3
docker                   /home/notoraptor/anaconda3/envs/docker
firebase                 /home/notoraptor/anaconda3/envs/firebase
graphcore                /home/notoraptor/anaconda3/envs/graphcore
myia                  *  /home/notoraptor/anaconda3/envs/myia

(myia) notoraptor@notoraptor-linux:~/mila/dev/git/myia/myia_backend_relay$ cat environment.yml 
name: myia_backend_relay
dependencies:
- python>=3.7,<4.0
- numpy>=1.16,<2.0
- scipy>=1.5,<2.0
- decorator>=4,<5
- attrs>=19,<20
- pip
- mila-iqia::tvm=0.7dev1+5.*
- pip:
  - antlr4-python3-runtime==4.7.2
(myia) notoraptor@notoraptor-linux:~/mila/dev/git/myia/myia_backend_relay$ conda env update --file environment.yml
Collecting package metadata (repodata.json): done
Solving environment: done
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Installing pip dependencies: / Ran pip subprocess with arguments:
['/home/notoraptor/anaconda3/envs/myia_backend_relay/bin/python', '-m', 'pip', 'install', '-U', '-r', '/media/win/Users/notoraptor/mila/dev/git/myia/myia_backend_relay/condaenv.jtiyxq2z.requirements.txt']
Pip subprocess output:
Processing /home/notoraptor/.cache/pip/wheels/63/93/52/480c17789fbf685f5d0e33e83c311f4a844cd6e2f5cf036c03/antlr4_python3_runtime-4.7.2-py3-none-any.whl
Installing collected packages: antlr4-python3-runtime
Successfully installed antlr4-python3-runtime-4.7.2

done
#
# To activate this environment, use
#
#     $ conda activate myia_backend_relay
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(myia) notoraptor@notoraptor-linux:~/mila/dev/git/myia/myia_backend_relay$ conda info --envs
# conda environments:
#
base                     /home/notoraptor/anaconda3
docker                   /home/notoraptor/anaconda3/envs/docker
firebase                 /home/notoraptor/anaconda3/envs/firebase
graphcore                /home/notoraptor/anaconda3/envs/graphcore
myia                  *  /home/notoraptor/anaconda3/envs/myia
myia_backend_relay       /home/notoraptor/anaconda3/envs/myia_backend_relay
```

Reply:
Ok